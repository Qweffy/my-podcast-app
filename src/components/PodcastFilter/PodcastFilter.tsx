import { PodcastFilterProps } from './PodcastFilter.types.ts'

export const PodcastFilter = ({ filter, setFilter }: PodcastFilterProps) => {
    return (
        <section className="flex justify-end w-3/5">
            <div className="flex gap-3 items-center">
                <span className="rounded-md font-bold bg-sky-600 text-white px-1 text">100</span>
                <input
                    type="text"
                    className="min-w-[14rem] rounded border-2 border-gray-100 shadow-sm text-sm px-3 py-2 placeholder font-semibold focus:shadow-xl outline-none ease-in-out duration-300 placeholder:text-gray-400"
                    placeholder="Filter podcasts..."
                    value={filter}
                    onChange={(e) => setFilter(e.target.value)}
                />
            </div>
        </section>
    )
}
