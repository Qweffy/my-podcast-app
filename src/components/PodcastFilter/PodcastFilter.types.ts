export interface PodcastFilterProps {
    filter: string
    setFilter: (filter: string) => void
}
