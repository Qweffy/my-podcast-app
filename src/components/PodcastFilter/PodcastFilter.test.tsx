import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import { describe, it, expect, vi } from 'vitest'
import PodcastFilter from 'components/PodcastFilter/index.ts'

describe('PodcastFilter', () => {
    it('should render correctly with default props', () => {
        const { container } = render(<PodcastFilter filter="" setFilter={() => {}} />)
        expect(container).toMatchSnapshot()
    })

    it('should handle empty filter value gracefully', () => {
        const setFilter = vi.fn()
        const { getByPlaceholderText } = render(
            <PodcastFilter filter="Initial" setFilter={setFilter} />,
        )
        const input = getByPlaceholderText('Filter podcasts...')
        fireEvent.change(input, { target: { value: '' } })
        expect(setFilter).toHaveBeenCalledWith('')
    })

    it('should handle non-empty filter value', () => {
        const setFilter = vi.fn()
        const { getByPlaceholderText } = render(<PodcastFilter filter="" setFilter={setFilter} />)
        const input = getByPlaceholderText('Filter podcasts...')
        fireEvent.change(input, { target: { value: 'New Filter' } })
        expect(setFilter).toHaveBeenCalledWith('New Filter')
    })
})
