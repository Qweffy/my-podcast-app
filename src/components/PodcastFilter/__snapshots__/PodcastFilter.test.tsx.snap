// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PodcastFilter > should render correctly with default props 1`] = `
<div>
  <section
    class="flex justify-end w-3/5"
  >
    <div
      class="flex gap-3 items-center"
    >
      <span
        class="rounded-md font-bold bg-sky-600 text-white px-1 text"
      >
        100
      </span>
      <input
        class="min-w-[14rem] rounded border-2 border-gray-100 shadow-sm text-sm px-3 py-2 placeholder font-semibold focus:shadow-xl outline-none ease-in-out duration-300 placeholder:text-gray-400"
        placeholder="Filter podcasts..."
        type="text"
        value=""
      />
    </div>
  </section>
</div>
`;
